app/src/main/java/org/mifos/mobile/ui/fragments/TransferProcessFragment.kt

```diff
@@ -49,42 +47,128 @@ class TransferProcessFragment : BaseFragment(), TransferProcessView {
-        }
-        binding.btnCancelTransfer.setOnClickListener {
-            cancelTransferProcess()
+        viewModel.transferUiState.observe(viewLifecycleOwner) {
+            when (it) {
+                is TransferUiState.Loading -> showProgress()
+                is TransferUiState.TransferSuccess -> {
+                    hideProgress()
+                    showTransferredSuccessfully()
+                }
+                is TransferUiState.Error -> {
+                    hideProgress()
+                    showError(MFErrorParser.errorMessage(it.errorMessage))
+                }
+            }
         }
-        binding.btnClose.setOnClickListener {
-            closeClicked()
+
+        with(binding) {
+            btnStartTransfer.setOnClickListener {
+                startTransfer()
+            }
+            btnCancelTransfer.setOnClickListener {
+                cancelTransferProcess()
+            }
+            btnClose.setOnClickListener {
+                closeClicked()
+            }
         }
     }
 
     /**
      * Initiates a transfer depending upon `transferType`
      */
-    fun startTransfer() {
+    private fun startTransfer() {
         if (!Network.isConnected(activity)) {
             Toaster.show(binding.root, getString(R.string.internet_not_connected))
             return
         }
+        val fromOfficeId = fromAccountOption?.officeId
+        val fromClientId = fromAccountOption?.clientId
+        val fromAccountType = fromAccountOption?.accountType?.id
+        val fromAccountId = fromAccountOption?.accountId
+        val toOfficeId = toAccountOption?.officeId
+        val toClientId = toAccountOption?.clientId
+        val toAccountType = toAccountOption?.accountType?.id
+        val toAccountId = toAccountOption?.accountId
+        val transferDate = DateHelper.getSpecificFormat(
+            DateHelper.FORMAT_dd_MMMM_yyyy,
+            getTodayFormatted(),
+        )
+        val transferAmount = binding.tvAmount.text.toString().toDouble()
+        val transferDescription = binding.tvRemark.text.toString()
+        var dateFormat: String = "dd MMMM yyyy"
+        var locale: String = "en"
+        val fromAccountNumber = fromAccountOption?.accountNo
+        val toAccountNumber = toAccountOption?.accountNo
         if (transferType == TransferType.SELF) {
-            presenter?.makeSavingsTransfer(payload)
+            viewModel.makeSavingsTransfer(
+                fromOfficeId,
+                fromClientId,
+                fromAccountType,
+                fromAccountId,
+                toOfficeId,
+                toClientId,
+                toAccountType,
+                toAccountId,
+                transferDate,
+                transferAmount,
+                transferDescription,
+                dateFormat,
+                locale,
+                fromAccountNumber,
+                toAccountNumber
+            )
         } else if (transferType == TransferType.TPT) {
```

Comment:
Why do we need to check for the `transferType` here? However, in the repository I see we are calling the same method for both of them. It is better to do this kind of checking in the data layer.